### PR TITLE
Fix #4788: Calendar typescript improvements

### LINKED ISSUE #4788: Calendar: CalendarChangeEvent type doesn't represent Calendar onChange event value when set to range
### Describe the bug

When I set the Calendar component to `selectionMode='range'` the types that come through on the `onChange` function `event.value` are: 
`Nullable<string | Date | Date[]>`
However, that means that I am not allowed to do: 
`[ new Date(), null ]`
Which seems to be a valid case when I have just one date selected and close the component:
image
image
(in this example I am updating the parent component onHide of the Calendar component, the currentValue is stored in state until this time)

I would like to suggest that the type `(Date | null)[]` be added as a valid type for the Calendar component onChange event. 

### Reproducer

_No response_

### PrimeReact version

9.6.1

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

_No response_

### Steps to reproduce the behavior

Build out the Calendar with `range` set as the selectionMode. 
Try to set the default value to something like `[ new Date(), null ]` (to mimic the Calendar in the single date select scenario)

### Expected behavior

Allow me to set defaults of the same type as those values that are used by the Calendar component.
Maybe even separate/default types for the different selectionModes.

### PR DESCRIPTION
**Pull request by melloware**
Fix #4788: Calendar typescript improvements

```ts
import React, { useState } from 'react';
import { Calendar} from 'primereact/calendar';
import { Nullable } from 'primereact/ts-helpers';

export default function RangeDemo() {
  const [dates, setDates] = useState<Nullable<(Date | null)[]>>(null);

  return (
    <div className="card flex justify-content-center">
      <Calendar
        value={dates}
        onChange={(e) => setDates(e.value)}
        selectionMode="range"
        readOnlyInput
      />
    </div>
  );
}
```

### CODE REVIEW

**Comment by melloware** [comment]
cc @bu3alwa

File: components/lib/calendar/calendar.d.ts [hunk 1]
```diff
@@ -269,19 +269,11 @@ export interface CalendarState {
...
+    viewDate: Nullable<Date>;
 }
 
-/**
- * Custom change event.
- * @see {@link CalendarProps.onChange}
- * @extends {FormEvent}
- * @event
- */
-interface CalendarChangeEvent extends FormEvent<Date | Date[] | string> {}
```
Comment (bu3alwa) [inline components/lib/calendar/calendar.d.ts:283]: Was string an improper type in this component?

File: components/lib/calendar/calendar.d.ts [hunk 2]
```diff
@@ -881,6 +858,62 @@ export interface CalendarProps {
     unstyled?: boolean;
 }
 
+interface CalendarPropsSingle extends CalendarBaseProps {
+    /**
+     * Specifies the selection mode.
```
Comment (bu3alwa) [inline components/lib/calendar/calendar.d.ts:863]: Maybe just minor comment improvements on these

File: components/lib/calendar/calendar.d.ts (hunk 1 above)
Comment (melloware) [reply components/lib/calendar/calendar.d.ts:283]: I could not find anywhere a String was used to represent the `value` anywhere in the code unless I am blind and missing something?

**Pull request merged**